compiler/qsc_codegen/src/qir_base.rs

```diff
@@ -474,12 +475,12 @@ impl Backend for BaseProfSim {
     fn qubit_allocate(&mut self) -> usize {
         let id = self.next_qubit_id;
         self.next_qubit_id += 1;
-        self.qubit_map.insert(id, id);
+        let _ = self.map(id);
         id
     }
 
     fn qubit_release(&mut self, _q: usize) {
-        // Base Profile qubits are never released, since they cannot be reused.
+        self.next_qubit_id -= 1;
```

Comment:
Are qubits guaranteed to be released in reverse order to allocation?

Reply:
Yes, as long as the caller is using qubit initialization statements, ie `use q = Qubit();`. We guarantee this by the way the `replace_qubit_allocation` pass works, since it creates an implied scope for each qubit init statement.